moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -0,0 +1,180 @@
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef MOVEIT_TRAJECTORY_PROCESSING_TIME_OPTIMAL_TRAJECTORY_GENERATION_H
+#define MOVEIT_TRAJECTORY_PROCESSING_TIME_OPTIMAL_TRAJECTORY_GENERATION_H
+
+#include <Eigen/Core>
+#include <list>
+#include <moveit/robot_trajectory/robot_trajectory.h>
+
+namespace trajectory_processing
+{
+class PathSegment
+{
+public:
+  PathSegment(double length = 0.0) : length(length)
+  {
+  }
+
+  virtual ~PathSegment()
+  {
+  }
+
+  double getLength() const
+  {
+    return length;
+  }
+  virtual Eigen::VectorXd getConfig(double s) const = 0;
+  virtual Eigen::VectorXd getTangent(double s) const = 0;
+  virtual Eigen::VectorXd getCurvature(double s) const = 0;
+  virtual std::list<double> getSwitchingPoints() const = 0;
+  virtual PathSegment* clone() const = 0;
+
+  double position;
+
+protected:
+  double length;
+};
+
+class Path
+{
+public:
+  Path(const std::list<Eigen::VectorXd>& path, double maxDeviation = 0.0);
+  Path(const Path& path);
+  ~Path();
+  double getLength() const;
+  Eigen::VectorXd getConfig(double s) const;
+  Eigen::VectorXd getTangent(double s) const;
+  Eigen::VectorXd getCurvature(double s) const;
+  double getNextSwitchingPoint(double s, bool& discontinuity) const;
+  std::list<std::pair<double, bool> > getSwitchingPoints() const;
+
+private:
+  PathSegment* getPathSegment(double& s) const;
+  double length;
+  std::list<std::pair<double, bool> > switchingPoints;
+  std::list<PathSegment*> pathSegments;
+};
+
+class Trajectory
+{
+public:
+  /// @brief Generates a time-optimal trajectory
+  Trajectory(const Path& path, const Eigen::VectorXd& maxVelocity, const Eigen::VectorXd& maxAcceleration,
```

Comment:
camelCase -> under_score

Reply:
One thing to consider: how far do we want to diverge from the upstream codebase? It makes merging new upstream stuff tougher. I'd suggest not changing variable names of the code that is imported from Tobias

Reply:
I'm curious - where is the upstream codebase?

Reply:
[tobiaskunz/trajectories](https://github.com/tobiaskunz/trajectories).

Reply:
Tobias hasn't updated that repository in 4 years, and he can't really do OSS anymore due to Waymo employment. Do we really think we'll be merging any new upstream stuff into here? The odds of that seem very low to me. 

If you want to keep the original formatting just to save effort, however, I guess I can see that.